cmd/root_test.go

```diff
@@ -29,100 +29,132 @@ import (
 	. "gopkg.in/check.v1"
 )
 
-func Test(t *testing.T) {
-	TestingT(t)
-}
-
 type MySuite struct{}
 
 var _ = Suite(&MySuite{})
 
-var storer *memory.Storage
-var fs billy.Filesystem
+func Test(t *testing.T) {
+	TestingT(t)
+}
 
 func TestMain(m *testing.M) {
-	GitInitialHash = "8fc4768edbef9b3f115a41eaf2a5740d41758cff"
-	storer = memory.NewStorage()
-	fs = memfs.New()
-
+	setup()
 	code := m.Run()
 	os.Exit(code)
 }
 
-func (m *MySuite) TestHpcToolkitRepo(c *C) {
-	_, callDir, _, _ := runtime.Caller(0)
-	expected := filepath.Dir(filepath.Dir(callDir))
+func setup() {
+}
 
-	// find repo when workdir is ./cmd subdir of repo
-	_, got, _ := hpcToolkitRepo()
-	c.Assert(got, Equals, expected)
+/* Tests */
+// root.go
+func (s *MySuite) TestHpcToolkitRepo(c *C) {
+	_, execPath, _, _ := runtime.Caller(0)
+	workDir := filepath.Dir(execPath)
+	repoDir := filepath.Dir(filepath.Dir(execPath))
 
-	// find repo when workdir is root of repo
-	os.Chdir(expected)
-	_, got, _ = hpcToolkitRepo()
-	c.Assert(got, Equals, expected)
+	// find repo when workdir is in ./cmd subdir of repo
+	_, got, err := hpcToolkitRepo()
+	c.Assert(got, Equals, workDir)
+	c.Assert(err, IsNil)
 
-	// find repo when workdir is outside root of repo, returns binary site in
-	// normal executions. Empty value expected during test since test binary is
-	// built in a test dir that is not in a git repository
-	os.Chdir(filepath.Dir(expected))
+	// find repo when workdir is root of repo
+	os.Chdir(repoDir)
+	_, got, err = hpcToolkitRepo()
+	c.Assert(got, Equals, repoDir)
+	c.Assert(err, IsNil)
+
+	// find repo when workDir is another git repo under ./hpc-toolkit
+	dir := ".." + "/hpc-toolkit/modules/network/vpc/.terraform/modules/cloud_router/.git"
+	err = os.Chdir(dir)
+	_, got, err = hpcToolkitRepo()
+	c.Assert(got, Equals, repoDir)
+	c.Assert(err, IsNil)
+
+	// Try to find repo when workdir is outside root of repo. Normal execution
+	// returns binary site in. Empty value expected during tests since test
+	// binary is built in a test dir that is not in a git repository
+	err = os.Chdir(filepath.Dir(repoDir))
 	_, got, _ = hpcToolkitRepo()
```

Comment:
nit: What is our defined api for when we can't find the repo. In the code you check for if err != nil, here you check that path is empty string. I would check for err in test instead of path.